src/config/cmd.rs

```diff
@@ -0,0 +1,113 @@
+use super::{load_builder_from_paths, load_source_from_paths, process_paths, ConfigPath};
+use crate::cli::handle_config_errors;
+use clap::Parser;
+use serde_json::Value;
+
+#[derive(Parser, Debug, Clone)]
+#[clap(rename_all = "kebab-case")]
+pub struct Opts {
+    /// Pretty print JSON
+    #[clap(short, long)]
+    pretty: bool,
+
+    /// Include default values where missing from config
+    #[clap(short, long)]
+    include_defaults: bool,
+}
+
+/// Helper to merge JSON. Handles objects and array concatenation.
+fn merge_json(a: &mut Value, b: Value) {
+    match (a, b) {
+        (Value::Object(ref mut a), Value::Object(b)) => {
+            for (k, v) in b {
+                merge_json(a.entry(k).or_insert(Value::Null), v);
```

Comment:
⚠️

One consequence to taking an entry in-place is that the resulting JSON field ordering when providing `--include_defaults` can differ, if the user config contains an environment string that is re-added after merge.

e.g.

```bash
➜  VECTOR_TEST_ADDRESS="127.0.0.1:8687" target/debug/vector -c ~/Desktop/vector-test.x config                            

{"api":{"address":"${VECTOR_TEST_ADDRESS}","enabled":true},"data_dir":"${VECTOR_DATA_DIR}","sinks":{"out_console":{"encoding":"text","inputs":["in_console"],"type":"console"}},"sources":{"in_console":{"type":"stdin"}}}
```

vs.

```bash
➜  VECTOR_TEST_ADDRESS="127.0.0.1:8687" target/debug/vector -c ~/Desktop/vector-test.x config --include-defaults

{"data_dir":"${VECTOR_DATA_DIR}","api":{"enabled":true,"address":"${VECTOR_TEST_ADDRESS}","playground":true},"datadog":null,"healthchecks":{"enabled":true,"require_healthy":false},"enrichment_tables":{},"sources":{"in_console":{"type":"stdin","max_length":102400,"host_key":null,"framing":{"method":"newline_delimited"},"decoding":{"codec":"bytes"}}},"sinks":{"out_console":{"inputs":["in_console","in_console"],"healthcheck_uri":null,"healthcheck":{"enabled":true,"uri":null},"buffer":{"type":"memory","max_events":500,"when_full":"block"},"type":"console","target":"stdout","encoding":"text"}},"transforms":{},"tests":[],"provider":null}
```

Notice that `api.enabled` and `api.address` swap positions.

This doesn't affect configuration hashing, since a stable-ordered representation with BTreeMap fields is used _before_ serialising.

I'm not sure that this warrants any additional work. The effect is negligible, and limited only to scenarios where defaults are merged (which is _not_ the case for Pipelines; those defaults will be coded in an upcoming schema and not part of reported configs.)

Just highlighting it as known drift.